tests/components/manual_mqtt/test_alarm_control_panel.py

```diff
@@ -11,1772 +10,1698 @@
-        ):
-            fire_time_changed(self.hass, future)
-            self.hass.block_till_done()
-        self.mock_publish.async_publish.assert_called_once_with(
-            "alarm/state", STATE_ALARM_ARMED_NIGHT, 0, True
-        )
-        self.mock_publish.async_publish.reset_mock()
+async def test_arm_home_no_pending(hass, mqtt_mock):
+    """Test arm home method."""
+    assert await async_setup_component(
+        hass,
+        alarm_control_panel.DOMAIN,
+        {
+            "alarm_control_panel": {
+                "platform": "manual_mqtt",
+                "name": "test",
+                "code": CODE,
+                "pending_time": 0,
+                "disarm_after_trigger": False,
+                "command_topic": "alarm/command",
+                "state_topic": "alarm/state",
+            }
+        },
+    )
+    await hass.async_block_till_done()
+
+    entity_id = "alarm_control_panel.test"
+
+    assert STATE_ALARM_DISARMED == hass.states.get(entity_id).state
+
+    await common.async_alarm_arm_home(hass, CODE)
+    await hass.async_block_till_done()
+
+    assert STATE_ALARM_ARMED_HOME == hass.states.get(entity_id).state
+
+
+async def test_arm_home_no_pending_when_code_not_req(hass, mqtt_mock):
+    """Test arm home method."""
+    assert await async_setup_component(
+        hass,
+        alarm_control_panel.DOMAIN,
+        {
+            "alarm_control_panel": {
+                "platform": "manual_mqtt",
+                "name": "test",
+                "code": CODE,
+                "code_arm_required": False,
+                "pending_time": 0,
+                "disarm_after_trigger": False,
+                "command_topic": "alarm/command",
+                "state_topic": "alarm/state",
+            }
+        },
+    )
+    await hass.async_block_till_done()
+
+    entity_id = "alarm_control_panel.test"
+
+    assert STATE_ALARM_DISARMED == hass.states.get(entity_id).state
+
+    await common.async_alarm_arm_home(hass, 0)
+    await hass.async_block_till_done()
+
+    assert STATE_ALARM_ARMED_HOME == hass.states.get(entity_id).state
+
+
+async def test_arm_home_with_pending(hass, mqtt_mock):
+    """Test arm home method."""
+    assert await async_setup_component(
+        hass,
+        alarm_control_panel.DOMAIN,
+        {
+            "alarm_control_panel": {
+                "platform": "manual_mqtt",
+                "name": "test",
+                "code": CODE,
+                "pending_time": 1,
+                "disarm_after_trigger": False,
+                "command_topic": "alarm/command",
+                "state_topic": "alarm/state",
+            }
+        },
+    )
+    await hass.async_block_till_done()
+
+    entity_id = "alarm_control_panel.test"
+
+    assert STATE_ALARM_DISARMED == hass.states.get(entity_id).state
+
+    await common.async_alarm_arm_home(hass, CODE, entity_id)
+    await hass.async_block_till_done()
+
+    assert STATE_ALARM_PENDING == hass.states.get(entity_id).state
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5745133)